<a href="https://colab.research.google.com/github/sherine9033/Machine-Learning/blob/main/Linear_Reg_multivariable.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In exercise folder (same level as this notebook on github) there is hiring.csv.

https://raw.githubusercontent.com/Rhevathi/DATA-SCIENCE/master/ML/2_linear_reg_multivariate/Exercise/hiring.csv This file contains hiring statics for a firm such as experience of candidate, his written test score and personal interview score. Based on these 3 factors, HR will decide the salary. Given this data, you need to build a machine learning model for HR department that can help them decide salaries for future candidates. Using this predict salaries for following candidates,

In [ ]:
import pandas as pd
import numpy as np
from sklearn import linear_model

In [ ]:
salary = pd.read_csv('https://raw.githubusercontent.com/Rhevathi/DATA-SCIENCE/master/ML/2_linear_reg_multivariate/Exercise/hiring.csv')
salary

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0        NaN                    8.0                           9      50000
1        NaN                    8.0                           6      45000
2       five                    6.0                           7      60000
3        two                   10.0                          10      65000
4      seven                    9.0                           6      70000
5      three                    7.0                          10      62000
6        ten                    NaN                           7      72000
7     eleven                    7.0                           8      80000

In [ ]:
salary['experience'] = salary['experience'].replace(np.NaN,0)

In [ ]:
salary['experience'] = salary['experience'].replace({'five':5,'two':2,'seven':7,'three':3,'ten':10,'eleven':11})

In [ ]:
salary['test_score(out of 10)'] = salary['test_score(out of 10)'].fillna(salary['test_score(out of 10)'].median())

In [ ]:
salary

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0           0                    8.0                           9      50000
1           0                    8.0                           6      45000
2           5                    6.0                           7      60000
3           2                   10.0                          10      65000
4           7                    9.0                           6      70000
5           3                    7.0                          10      62000
6          10                    8.0                           7      72000
7          11                    7.0                           8      80000

In [ ]:
# Create linear regression object
reg = linear_model.LinearRegression()
reg.fit(salary.drop('salary($)',axis='columns'),salary['salary($)'])

LinearRegression()

In [ ]:
reg.coef_

array([2812.95487627, 1845.70596798, 2205.24017467])

In [ ]:
reg.intercept_

17737.263464337688

# **Salary of candidate with 2 yr experience, 9 test score, 6 interview score**

In [ ]:
reg.predict([[2, 9, 6]])

/usr/local/lib/python3.8/dist-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([53205.96797671])

In [ ]:
2812.95487627*2 + 1845.70596798*9 + 2205.24017467*6 + 17737.263464337688

53205.96797671769

**Answer:** 53205.96797671

# **Salary of candidate with 12 yr experience, 10 test score, 10 interview score**

In [ ]:
reg.predict([[12, 10, 10]])

/usr/local/lib/python3.8/dist-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([92002.18340611])

**Answer:** 92002.18340611